rust/agama-lib/src/files/model.rs

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) [2024] SUSE LLC
+//
+// All Rights Reserved.
+//
+// This program is free software; you can redistribute it and/or modify it
+// under the terms of the GNU General Public License as published by the Free
+// Software Foundation; either version 2 of the License, or (at your option)
+// any later version.
+//
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+// more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, contact SUSE LLC.
+//
+// To contact SUSE LLC about this file by physical or electronic mail, you may
+// find current contact information at www.suse.com.
+
+//! Implements a data model for Files configuration.
+
+use serde::{Deserialize, Serialize};
+use std::fs::OpenOptions;
+use std::os::unix::fs::OpenOptionsExt;
+use std::io::Write;
+
+use crate::{error::ServiceError, utils::Transfer};
+
+#[derive(Clone, Debug, Serialize, Deserialize, utoipa::ToSchema)]
+#[serde(untagged)]
+pub enum FileSource {
+    /// File body directly written
+    Text { content: String },
```

Comment:
If this one is called "content", which looks correct to me, we should change the scripts to use the same naming.

Reply:
yeah, I do not like much "body" which looks a bit strange in context of file. I see more common to talk about file content and not file body ( as it emphasize some headers.). So if you agree, I can change here also script for identical name...should we still keep backward compatibility for scripts or not?

Reply:
Yes, it makes sense to change it to "content" but, if possible, we should be backward compatible (at least for a while).